### PR TITLE
[iam] Implement iam cpp

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

**Review by mlohvynenko — APPROVED** [review]

File: recipes-aos/aos-iamanager/files/aos-iamanager-provisioning.service [hunk 1]
```diff
@@ -7,7 +7,7 @@ PartOf=aos.target
 
 [Service]
 Type=notify
-ExecStart=/usr/bin/aos_iamanager -c /etc/aos/aos_iamanager.cfg -provisioning -v debug -j
```
Comment (al1img) [inline recipes-aos/aos-iamanager/files/aos-iamanager-provisioning.service:10]: Keep `-c /etc/aos/aos_iamanager.cfg`

File: recipes-aos/aos-iamanager/files/aos-iamanager.service [hunk 2]
```diff
@@ -7,7 +7,7 @@ PartOf=aos.target
 
 [Service]
 Type=notify
-ExecStart=/usr/bin/aos_iamanager -c /etc/aos/aos_iamanager.cfg -v debug -j
```
Comment (al1img) [inline recipes-aos/aos-iamanager/files/aos-iamanager.service:10]: ditto

File: recipes-connectivity/openssl/openssl_%.bbappend [hunk 3]
```diff
@@ -0,0 +1,6 @@
+do_install:append () {
+    sed -i '/\[engine_section\]/d' ${D}${sysconfdir}/ssl/openssl.cnf
+    sed -i '/^pkcs11 =/d' ${D}${sysconfdir}/ssl/openssl.cnf
+    echo "[engine_section]" >> ${D}${sysconfdir}/ssl/openssl.cnf
+    echo "pkcs11 = pkcs11_section" >> ${D}${sysconfdir}/ssl/openssl.cnf
+}
```
Comment (mykola-kobets-epam) [inline recipes-connectivity/openssl/openssl_%.bbappend:6]: please add
"dynamic_path = path to libpkcs11.so\n"
"MODULE_PATH = path to softhsm lib\n"
"init = 0\n"

File: recipes-aos/aos-iamanager/aos-iamanager_git.bb [hunk 4]
```diff
@@ -17,69 +13,31 @@ SRC_URI += " \
     file://aos-target.conf \
 "
 
-inherit go goarch systemd
+PARALLEL_MAKE = "-j1"
+PARALLEL_MAKEINST = "-j1"
 
-SYSTEMD_SERVICE:${PN} = "aos-iamanager.service aos-iamanager-provisioning.service"
+DEPENDS += "poco systemd grpc grpc-native protobuf-native protobuf openssl libp11"
```
Comment (al1img) [inline recipes-aos/aos-iamanager/aos-iamanager_git.bb:19]: libp11 should be add to RDEPENDS instead.

File: recipes-connectivity/openssl/openssl_%.bbappend [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
```
Comment (al1img) [inline recipes-connectivity/openssl/openssl_%.bbappend:1]: Add empty line after FILESEXTRAPATHS

File: recipes-aos/aos-iamanager/aos-iamanager_git.bb [hunk 6]
```diff
@@ -17,69 +13,29 @@ SRC_URI += " \
...
-    echo 'import (' >> ${file}
-
-    for module in ${AOS_IAM_IDENT_MODULES}; do
-        echo "\t_ \"${GO_IMPORT}/${module}\"" >> ${file}
-    done
-
-    echo ')' >> ${file}
-}
-
-do_fetch[vardeps] += "AOS_NODE_ID AOS_NODE_TYPE AOS_IAM_NODES AOS_IAM_HOSTNAMES"
```
Comment (al1img) [inline recipes-aos/aos-iamanager/aos-iamanager_git.bb:82]: Keep this line

File: recipes-aos/aos-iamanager/files/aos-iamanager-provisioning.service (hunk 1 above)
Comment (MykolaSolyanko) [reply recipes-aos/aos-iamanager/files/aos-iamanager-provisioning.service]: Done

File: recipes-aos/aos-iamanager/files/aos-iamanager.service (hunk 2 above)
Comment (MykolaSolyanko) [reply recipes-aos/aos-iamanager/files/aos-iamanager.service]: Done

File: recipes-aos/aos-iamanager/files/aos-iamanager-provisioning.service (hunk 1 above)
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply recipes-aos/aos-iamanager/files/aos-iamanager-provisioning.service]: Done

File: recipes-connectivity/openssl/openssl_%.bbappend (hunk 5 above)
Comment (MykolaSolyanko) [reply recipes-connectivity/openssl/openssl_%.bbappend:1]: Done

File: recipes-aos/aos-iamanager/aos-iamanager_git.bb (hunk 6 above)
Comment (MykolaSolyanko) [reply recipes-aos/aos-iamanager/aos-iamanager_git.bb:82]: Done

File: recipes-aos/aos-iamanager/aos-iamanager_git.bb [hunk 7]
```diff
@@ -1,14 +1,10 @@
...
+DESCRIPTION = "AOS Identity and Access Manager CPP" 
 LICENSE = "Apache-2.0"
-LIC_FILES_CHKSUM = "file://src/${GO_IMPORT}/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57"
-
-BRANCH = "develop"
-SRCREV = "${AUTOREV}"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57"
 
-SRC_URI = "git://${GO_IMPORT}.git;branch=${BRANCH};protocol=https"
+BRANCH = "integration"
```
[the remark this answers is not in the document]
Comment (al1img) [reply recipes-aos/aos-iamanager/aos-iamanager_git.bb:5]: keep develop branch

**Pull request merged**